toolkit/tools/scheduler/scheduler.go

```diff
@@ -457,6 +476,53 @@ func updateGraphWithImplicitProvides(res *schedulerutils.BuildResult, pkgGraph *
 	return
 }
 
+// setAssociatedDeltaPaths sets the DeltaPath for all of the request's ancillary nodes (both build and run) to the actual
+// RPM paths. A delta node will normally point at the cached RPM path, but we want to point it at the actual RPM if we built it.
+func setAssociatedDeltaPaths(res *schedulerutils.BuildResult, builtFiles []string, pkgGraph *pkggraph.PkgGraph, graphMutex *sync.RWMutex) (err error) {
+	graphMutex.Lock()
+	defer graphMutex.Unlock()
+
+	if !res.WasDelta {
+		err = fmt.Errorf("setAssociatedDeltaPaths called on non-delta node: %s", res.Node.FriendlyName())
+		return
+	}
+
+	// Build map of basename to full path for built files. This will allow us to find the actual RPM path we built for
+	// any given .rpm file built from our ancillary nodes.
+	builtFileMap := make(map[string]string)
+	for _, builtFile := range builtFiles {
+		// We should not expect to see multiple built files with the same basename
+		_, hasConflict := builtFileMap[filepath.Base(builtFile)]
```

Comment:
>filepath.Base(builtFile) [](http://example.com/codeflow?start=33&length=24)

Nit-pick: we're using `filepath.Base(builtFile)` 3 times in this scope. Let's safe it to a variable.

Reply:
Done